lte/gateway/c/core/oai/tasks/mme_app/mme_app_detach.c

```diff
@@ -201,11 +202,20 @@ void mme_app_handle_detach_req(const mme_ue_s1ap_id_t ue_id) {
       }
     }
   } else {
+    // delete gtpv2c tunnel on last PDN
+    bool no_delete       = false;
+    pdn_cid_t cid_delete = 0;
+    for (pdn_cid_t i = 0; i < MAX_APN_PER_UE; i++) {
+      if (ue_context_p->pdn_contexts[i]) {
+        cid_delete = i;
+      }
+    }
     for (pdn_cid_t i = 0; i < MAX_APN_PER_UE; i++) {
       if (ue_context_p->pdn_contexts[i]) {
         // Send a DELETE_SESSION_REQUEST message to the SGW
         mme_app_send_delete_session_request(
-            ue_context_p, ue_context_p->pdn_contexts[i]->default_ebi, i);
+            ue_context_p, ue_context_p->pdn_contexts[i]->default_ebi, i,
+            (cid_delete != i));
```

Comment:
we should update `no_delete` bool and use it here?

Reply:
yeah, was just updating those. I am changing couple of more things. I will ping you :)